aws-transfer-agreement/aws-transfer-agreement.json

```diff
@@ -0,0 +1,155 @@
+{
+  "typeName": "AWS::Transfer::Agreement",
+  "description": "Resource Type definition for AWS::Transfer::Agreement",
+  "definitions": {
+    "Tag": {
+      "description": "Creates a key-value pair for a specific resource.",
+      "type": "object",
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The name assigned to the tag that you create.",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "Value": {
+          "type": "string",
+          "description": "Contains one or more values that you assigned to the key name you create.",
+          "minLength": 0,
+          "maxLength": 256
+        }
+      },
+      "required": [
+        "Key",
+        "Value"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "properties": {
+    "Description": {
+      "description": "A textual description for the agreement.",
+      "type": "string",
+      "pattern": "^[\\w\\- ]*$",
```

Comment:
I think we should be using this description https://code.amazon.com/packages/NeccoAs2CommonModel/blobs/5d17f26ad01ca95d9cab10acf1340b1e66709411/--/model/common-type.xml#L92,L93,L94,L95,L96,L97,L98,L99 for our AS2 description.

Reply:
Oh good catch, you're right

Reply:
Actually diving deeper in the regex, what we have here is equivalent to the one in the model. I matches every character plus space.

Reply:
Oh nice! We might want to make sure we have the right max and min values though since it looked different to workflows.